### ISSUE TITLE
bug: switchNetwork always uses public rpcUrl when adding network to wallet

### ISSUE THREAD

**Issue by thomasvalera** [wagmi-dev/wagmi#2810]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.3.9

### Current Behavior

Using `switchNetwork` always adds the network to my wallet (MetaMask in this case) using the public rpcUrl instead of the default. 

This is a real problem as the mumbai testnet's public rpcUrl doesn't seem to work anymore so it has become impossible to develop my dapp using wagmi!

Here is my config:

```
const { chains, publicClient } = configureChains(
  [polygonMumbai],
  [
    jsonRpcProvider({
      rpc: (chain) => ({
        http: `https://rpc-mumbai.maticvigil.com`,
      }),
    }),
  ],
);

const connectors = connectorsForWallets([
  {
    groupName: "Recommended",
    wallets: [
      metaMaskWallet({
        projectId: walletConnectProjectId,
        chains,
        shimDisconnect: false,
      }),
    ],
  },
]);

const config = createConfig({
  autoConnect: true,
  connectors,
  publicClient,
});

export default function WagmiProvider(props: Props): JSX.Element {
  return (
    <WagmiConfig config={config}>
      <RainbowKitProvider chains={chains}> {props.children}</RainbowKitProvider>
    </WagmiConfig>
  );
}
```

And to switch network I use:

```
const { switchNetwork, chains: sChains } = useSwitchNetwork({
    chainId: polygonMumbai.id,
    onSuccess: (data, args) => {},
    onError: () => {},
  });
  
  switchNetwork?.();
```

This gives the following error when approving the new network in MetaMask:
```
Error while connecting to the custom network.
```

### Expected Behavior

When adding a new network, it should take the default network specified using configureChains.

### Steps To Reproduce

The error also occurs when adding the network using mumbai.polygonscan

1. Remove the mumbai network from your MetaMask wallet
2. Configure a custom RPC provider in `configureChains` for the mumbai network
3. Call "switchNetwork" function from the "useSwitchNetwork" react hook
4. This should open a "add network" popup in your MetaMask with the RPC url "https://matic-mumbai.chainstacklabs.com"
5. Hit approve
6. Wait for the RPC to timeout

**Comment by HT-Moh** [comment]
Yes this is true the [matic-mumbai.chainstacklabs.com](https://matic-mumbai.chainstacklabs.com/) it's not working anymore I think it has been replaced with `https://rpc-mumbai.maticvigil.com/` 

![image](https://github.com/wagmi-dev/wagmi/assets/34759569/16c66b3a-b878-4fbc-9320-0b046e8eeb8b)

**Comment by Nachobr** [comment]
This is happening to us too, any fixes or temporary ?

**Comment by thomasvalera** [comment]
@Nachobr I haven't fix the part where it adds the network with the incorrect rpcUrl unfortunately.
BUT if you add a new provider to `configureChains` AND add the same network manually to MetaMask you're able to switch and make calls without it failing.
For now that will have to do.

```
const { chains, publicClient } = configureChains(
  [polygonMumbai],
  [
    jsonRpcProvider({
      rpc: (chain) => ({
        http: `https://rpc-mumbai.maticvigil.com`,
      }),
    }),
  ],
);
```

And add the network manually in MetaMask (or other wallet I suppose):
```
Network name: Mumbai
Network URL: https://rpc-mumbai.maticvigil.com
Chain ID: 80001
Currency symbol: MATIC
```

**Comment by tmm** [comment]
I'd argue that public RPCs tend to be more stable than developer-provided RPC URLs that are often tied to personal or organization accounts. In this case, we should update the chain to use a more stable public RPC.

**Comment by HT-Moh** [comment]
Workaround which worked is to use Alchemey provider and remove the public provider

**Comment by Lrodriguez9719** [comment]
@thomasvalera How did you added the network manually in MetaMask?
